### ISSUE TITLE
Postgres agent not displaying any elements in UI

### ISSUE THREAD

**Issue by ReidWeb** [librenms/librenms-agent#98]
The postgresql agent is not displaying any graphs in the UI under 'applications->postgresql' under device.

![img](https://i.imgur.com/fn0MfeE.png)

When viewing the graphs under 'apps', it shows as blank graphs

![img2](https://i.imgur.com/lPCMTSP.png)

Graphs for other applications correctly display data.

I have added

```
extend postgres /etc/snmp/postgres
```

to the `snmpd.conf` and restarted the `snmpd` service.

When I manually run `/etc/snmp/postgres` I get the correct and expected output of postgresql stats.

When I manually invoke a poll of the host using `poller.php` I notice the output does not include `postgresql`:

```
#### Load poller module applications ####
 nginx mysql

>> Runtime for poller module 'applications': 0.9013 seconds with 104376 bytes
#### Unload poller module applications ####
```

Running `./validate.php` returns
```
==========================================================
Component | Version
--------- | -------
LibreNMS  | ead3abb2c259de64fdaf592e806127bf546c23b9
DB Schema | 173
PHP       | 5.6.30-0+deb8u1
MySQL     | 5.5.54-0+deb8u1
RRDTool   | 1.4.8
SNMP      | NET-SNMP 5.7.2.1
==========================================================

[WARN]  Your install is out of date, last update: Sun, 05 Mar 2017 19:42:39 +0000
[OK]    Database connection successful
[FAIL]  You have no timezone set for php. [FIX] http://php.net/manual/en/datetime.configuration.php#ini.date.timezone
```

The timezone fail issue is strange - but the system functions fine without it, I've tried fixing it by setting the value and restarting `php-fpm` but it did not resolve issue - don't believe that to be related to this issue.

**Comment by VVelox** [comment]
What is the output of /etc/snmp/postgres and 'check_postgres.pl -u $DBuser --action dbstats'?

**Comment by ReidWeb** [comment]
Output of /etc/snmp/postgres is

```
8
433911163
12595
277172
8.26121e+09
0
0
0
0
0
0
0
1.059e+11
3.21839e+09
21623870
73463423
21306336
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 template1
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 template0
1 959128 22 1487 27374687 0 0 0 0 0 0 0 438252090 5303787 0 273 0 postgres
20 86492241 6024 28292 996600796 0 0 0 0 0 0 0 7414790685 304450690 16161771 23937194 16133319 a_db
8 58990013 4940 178972 4317498769 0 0 0 0 0 0 0 91312921845 1199823367 4601599 3014516 4366572 b_db
1 204173727 98 13474 381069811 0 0 0 0 0 0 0 2162270869 148170202 236566 17470900 225003 c_db
0 8071603 222 3001 549360327 0 0 0 0 0 0 0 1417032172 266230245 8307 3783 1656 d_db
3 60410859 125 6843 1934640052 0 0 0 0 0 0 0 2899208530 1273103187 29013 29000382 27099 e_db
4 15228999 1186 44227 64875133 0 0 0 0 0 0 0 433229505 23063775 581654 32714 549440 f_db
2 543721 0 2363 17168143 0 0 0 0 0 0 0 260137614 3545251 4960 3934 3247 g_db
```

I have to run `check_postgres.pl` providing a password for it to work

```
/usr/local/nagios/postgres/check_postgres.pl -u librenms_monitor -dbpass mypassword --dbhost localhost  --port 5432 --action dbstats
```

returns

```
backends:0 commits:0 rollbacks:0 read:0 hit:0 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:0 fetch:0 ins:0 upd:0 del:0 dbname:template1
backends:0 commits:0 rollbacks:0 read:0 hit:0 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:0 fetch:0 ins:0 upd:0 del:0 dbname:template0
backends:1 commits:959144 rollbacks:22 read:1487 hit:27376731 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:438262026 fetch:5304763 ins:0 upd:273 del:0 dbname:postgres
backends:20 commits:86493303 rollbacks:6024 read:28292 hit:996612747 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:7414876841 fetch:304454069 ins:16161993 upd:23937441 del:16133541 dbname:jira_db
backends:9 commits:58990436 rollbacks:4940 read:178973 hit:4317578384 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:91315254729 fetch:1199825132 ins:4601659 upd:3014542 del:4366632 dbname:confluence_db
backends:3 commits:204177287 rollbacks:98 read:13475 hit:381078314 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:2162295786 fetch:148174413 ins:236579 upd:17471058 del:225016 dbname:bitbucketdb
backends:0 commits:8071611 rollbacks:222 read:3001 hit:549360735 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:1417041788 fetch:266230289 ins:8307 upd:3783 del:1656 dbname:bamboodb
backends:3 commits:60411514 rollbacks:125 read:6843 hit:1934662034 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:2899242646 fetch:1273115899 ins:29013 upd:29000718 del:27099 dbname:crowddb
backends:4 commits:15229257 rollbacks:1186 read:44227 hit:64876452 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:433236946 fetch:23064410 ins:581654 upd:32714 del:549440 dbname:sonarqube_db
backends:2 commits:543729 rollbacks:0 read:2363 hit:17168419 idxscan:0 idxtupread:0 idxtupfetch:0 idxblksread:0 idxblkshit:0 seqscan:0 seqtupread:0 ret:260142390 fetch:3545295 ins:4960 upd:3934 del:3247 dbname:ghost_db
```

**Comment by VVelox** [comment]
What version of awk do you have installed? It should not be giving you stuff like 3.21839e+09. Honestly not even sure if RRD will process that.

**Comment by ReidWeb** [comment]
Awk version
```
awk -Wversion 2>/dev/null || awk --version
mawk 1.3.3 Nov 1996, Copyright (C) Michael D. Brennan
```

**Comment by ReidWeb** [comment]
Installed GNU awk (GNU awk), now awk version prints

```
awk -Wversion 2>/dev/null || awk --version
GNU Awk 4.1.1, API: 1.1 (GNU MPFR 3.1.2-p3, GNU MP 6.0.0)
Copyright (C) 1989, 1991-2014 Free Software Foundation.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see http://www.gnu.org/licenses/.
```

**Comment by VVelox** [comment]
Well the good news is it does produce the proper output using mawk(atleast it does with 1.3.4) when installed it and changed it to use that.

[root@vixen42]/arc/src/ports/lang/mawk# mawk -Wversion
mawk 1.3.4 20161120
Copyright 2008-2015,2016, Thomas E. Dickey
Copyright 1991-1996,2014, Michael D. Brennan

random-funcs:       arc4random_stir/arc4random
regex-funcs:        internal
compiled limits:
sprintf buffer      8192
maximum-integer     2147483647
[root@vixen42]/arc/src/ports/lang/mawk# /root/snmp-extends/postgres
6
826278
68
4090
30804033
0
0
0
0
0
0
0
237828633
11700479
14735
8649
14698
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 template0
2 10944827 1 502 35765466 0 0 0 0 0 0 0 1113217115 2945691 0 0 0 postgres
0 394520 68 1056 18320244 0 0 0 0 0 0 0 42419915 8907502 12406 8032 12404 roundcube
6 144294 0 1299 3723502 0 0 0 0 0 0 0 43984067 1124864 2329 617 2294 rt
0 50626 0 279 1546735 0 0 0 0 0 0 0 28614142 280153 0 0 0 webcalendar
0 50626 0 288 1546769 0 0 0 0 0 0 0 27121600 280179 0 0 0 northamerica
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 template1
0 69428 0 528 2352076 0 0 0 0 0 0 0 43602609 428974 0 0 0 davical
0 66158 0 367 1869162 0 0 0 0 0 0 0 27595732 398654 0 0 0 monitoringtest

**Comment by VVelox** [comment]
I would begin scouring your environmental and locale settings to see if you have any settings that may cause that.

**Comment by ReidWeb** [comment]
@VVelox See my last comment? Installing GNU awk seems to have resolved the exponential value issue?

**Comment by ReidWeb** [comment]
Graphs for postgres are now showing data under the generic 'Apps' view

![img](https://i.imgur.com/w7biAWL.png)

but are still not showing in the apps tab of the device

**Comment by VVelox** [comment]
hmm... appears that html/pages/device/apps/postgres.inc.php was not part of pull request  #6004

**Comment by VVelox** [comment]
https://github.com/librenms/librenms/pull/6110/

That has the fix.

**Comment by ReidWeb** [comment]
Issue has been fixed now with librenms/librenms#6110

Thanks for the quick turn around @VVelox !
